### PR TITLE
[C-1654] Only show individual tracks on offline tracks tab

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description

Tracks that were downloaded as part of a favorited collection via the favorites toggle were showing up in the tracks tab lineup. The tracks tab should only show individual tracks.

### Feature Flags ###

`OFFLINE_MODE_ENABLED`

### CODE REVIEW

File: packages/common/src/models/Track.ts [hunk 1]
```diff
@@ -130,9 +130,14 @@ export type TrackMetadata = {
...
 
+export type DownloadReason = {
+  is_from_favorites?: boolean
+  collection_id?: string
+}
+
 // This is available on mobile for offline tracks
 export type OfflineTrackMetadata = {
-  downloaded_from_collection: string[]
+  reasons_for_download: DownloadReason[]
```
Comment (dylanjeffers) [inline packages/common/src/models/Track.ts:140]: This is fun

File: packages/mobile/src/components/offline-downloads/DownloadToggle.tsx [hunk 2]
```diff
@@ -88,11 +94,11 @@ export const DownloadToggle = ({
   const offlineDownloadStatus = useSelector(getOfflineDownloadStatus)
   const isAnyDownloadInProgress = useMemo(
     () =>
-      trackIds.some((trackId: number) => {
+      tracksForDownload.some(({ trackId }: TrackForDownload) => {
```
Comment (dylanjeffers) [inline packages/mobile/src/components/offline-downloads/DownloadToggle.tsx:97]: I think you don't need to include type here since array should be typed?

**Review by dylanjeffers — COMMENTED** [review]
Stopping full review short, but wanted to at least post comments I have so far

File: packages/mobile/src/components/offline-downloads/OfflineDownloader.tsx [hunk 3]
```diff
@@ -13,5 +14,7 @@ export const OfflineDownloader = () => {
     }
   }, [initialized, isOfflineModeEnabled])
 
+  useLoadOfflineTracks()
```
Comment (Kyle-Shanks) [inline packages/mobile/src/components/offline-downloads/OfflineDownloader.tsx:17]: ah nice, this will cover the issue that i had 👍

**Review by Kyle-Shanks — APPROVED** [review]
Nice

**Review by dylanjeffers — APPROVED** [review]
lgtm

**Pull request merged**